precompiles/bank/bank.go

```diff
@@ -0,0 +1,130 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package bank
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+	cmn "github.com/evmos/evmos/v15/precompiles/common"
+	erc20keeper "github.com/evmos/evmos/v15/x/erc20/keeper"
+)
+
+// PrecompileAddress defines the bank precompile address in Hex format
+const PrecompileAddress = "0x0000000000000000000000000000000000000804"
+
+var _ vm.PrecompiledContract = &Precompile{}
+
+// Embed abi json file to the executable binary. Needed when importing as dependency.
+//
+//go:embed abi.json
+var f embed.FS
+
+type Precompile struct {
+	cmn.Precompile
+	bankKeeper  bankkeeper.Keeper
+	erc20Keeper erc20keeper.Keeper
+}
+
+// NewPrecompile creates a new bank Precompile instance as a
+// PrecompiledContract interface.
+func NewPrecompile(
+	bankKeeper bankkeeper.Keeper,
+	erc20Keeper erc20keeper.Keeper,
+) (*Precompile, error) {
+	abiBz, err := f.ReadFile("abi.json")
+	if err != nil {
+		return nil, err
+	}
+
+	newAbi, err := abi.JSON(bytes.NewReader(abiBz))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Precompile{
+		Precompile:  cmn.Precompile{ABI: newAbi},
+		bankKeeper:  bankKeeper,
+		erc20Keeper: erc20Keeper,
+	}, nil
+}
+
+// Address defines the address of the bank compile contract.
+// address: 0x0000000000000000000000000000000000000804
+func (Precompile) Address() common.Address {
+	return common.HexToAddress(PrecompileAddress)
+}
+
+// IsStateful returns false since the precompile contract doesn't perform any
+// state transitions.
+func (Precompile) IsStateful() bool {
```

Comment:
We removed these methods from the other precompiles so no need to add it here

Reply:
We still have it inside outposts precompiles. Should be removed?